biscuit.go

```diff
@@ -432,6 +455,13 @@ func (b *Biscuit) BlockCount() int {
 	return len(b.container.Blocks)
 }
 
+func (b *Biscuit) RootKeyID() (uint32, bool) {
+	if v := b.container.RootKeyId; v != nil {
+		return *v, true
+	}
+	return 0, false
+}
```

Comment:
would returning a pointer instead of a tuple also work to signal optionality?

Reply:
I vacillated over this decision for a while before implementing this new method. I'm glad you asked about it.

Yes, it would work to return a pointer, though it's fairly typical in Go to use this style, especially when the primary return value's type has a sensible _zero value_.

This paired value style comes from and is adapted somewhat unjustly from a few of Go's statements and expressions that benefit from it:
- [Channel receive statements](https://go.dev/ref/spec#RecvStmt) that can include an optional second untyped boolean value to check whether the channel is closed.
- [Map index expressions](https://go.dev/ref/spec#Index_expressions) that can use an optional second untyped boolean value to check whether the given key is present in the map.
- [Type assertions](https://go.dev/ref/spec#Type_assertions) can use an optional second untyped boolean value to check whether the interface value's dynamic type matches the asserted type.

In each of these cases, callers are free to omit the assignment of this second value, and I don't mean just not assigning it to a named variable. Callers can omit mention of the second value entirely if they don't care about it. By contrast, if a map index expression returned by pointer to indicate whether or not the key was bound to a value in the map, all callers would have to confront the possibility of absence.

My use of that idiom here doesn't benefit from that same treatment, though. A caller of my `(*Biscuit).RootKeyID` cannot ignore _the presence_ of that second return value. They _are_ free to ignore _its value_, though, and use the first value unconditionally, since the zero value for `uint32` is, well, _zero_, which is a valid key ID.

Given that, I'm now coming back around to thinking that returning by pointer would be a safer choice here. Please allow me to make that change, and then you can decide if you like that by-pointer approach better.

Reply:
i also think returning a pointer would be a bit safer. The rust implementation distinguishes between no id and 0. Since biscuit uses protobuf 2, I think the serialization also makes that distinction.

Reply:
Please see commit 9b7f488f56eebf7d412241f9563d635b348401c8 for this change. I'll squash the commits before merging.

Reply:
looks good, thanks! in can also squash-merge the PR if that’s simpler for you

Reply:
Yes, that's fine with if you squash the commits. I recommend preserving the message from the first one.

After we merge this, I'll carry on with addressing the rest of #150—selecting a public key in calls to the `(b *Biscuit).Authorizer` method, or a new sibling method, per https://github.com/biscuit-auth/biscuit-go/issues/150#issuecomment-2444286036.